lib/src/components/slider/slider.dart

```diff
@@ -113,25 +117,49 @@ class _SliderPainter extends CustomPainter {
       ..style = PaintingStyle.stroke
       ..strokeWidth = 1;
 
-    final RRect track = RRect.fromLTRBR(0.0, 12.0, size.width, size.height - 12, const Radius.circular(12.0));
-    final RRect active = RRect.fromLTRBR(
-      0.0,
-      12.0,
-      (size.width * value) > 24 ? (size.width * value) : 24.0,
-      size.height - 12.0,
-      const Radius.circular(12.0),
-    );
+    final Paint dividerPaint = Paint()
+      ..color = outlineColor
+      ..style = PaintingStyle.fill;
 
-    final Offset thumbPosition = Offset(
-      (size.width * value) > 24 ? (size.width * value) - ((size.height - 26) / 3) - 6 : 24 - 12,
-      size.height / 2,
-    );
+    double activeWidth() {
+      if (size.width * value >= size.width) {
+        return size.width;
+      } else if (size.width * value < 24) {
+        return 24;
+      } else {
+        return size.width * value + 12;
+      }
+    }
+
+    double thumbPositionX() {
+      if (size.width * value >= size.width) {
+        return size.width - 12;
+      } else if (size.width * value < 24) {
+        return 12;
+      } else {
+        return size.width * value;
+      }
+    }
```

Comment:
I believe a switch case would not work there as size is not constant

Reply:
using gates this can work easily, tho to simplify just make `return ((size.width - 24.0) * value.clamp(0.0, 1.0)) + 12.0;`, same thing for the other one